### DISCUSSION TITLE
Feature Request: Separate the number of threads (-r) settings between phases

### DISCUSSION THREAD

**Discussion by archerallstars** [madMAx43v3r/chia-plotter#562] [Ideas]
The reason are my results with AMD 5950X, `-u` and `-v` = 512, `-t` = Corsair MP600 (non RAID), `-2` = DDR4-2666 110G::

1. **Number of Threads: 30**
```
Phase 1 took 601.129 sec
Phase 2 took 387.489 sec
Phase 3 took 319.588 sec
Phase 4 took 37.1284 sec
Total plot creation time was 1345.38 sec (22.423 min)
```

2. **Number of Threads: 20**
```
Phase 1 took 585.479 sec
Phase 2 took 392.958 sec
Phase 3 took 306.668 sec
Phase 4 took 33.5924 sec
Total plot creation time was 1318.75 sec (21.9791 min)
```

3. **Number of Threads: 16**
```
Phase 1 took 584.581 sec
Phase 2 took 401.289 sec
Phase 3 took 301.141 sec
Phase 4 took 32.9146 sec
Total plot creation time was 1319.98 sec (21.9996 min)
```

It seems that throwing as many threads as possible would only help in _**phase 2**_, while it would be better to limit the number of threads to the number of physical cores in the other phases.

**Comment by farchord** [comment]
This is a personal opinion, but it seems to be a difference of about 2%... Not sure it's that important...

Instead, maybe give a Thread option, and a Core option? i.e. in some phases, if it benefits more from the thread count then use that, otherwise use the core count?

But at that point it's really just tiny optimizations...
